kernel/src/main/java/org/kframework/kast/KastFrontEnd.java

```diff
@@ -124,13 +135,59 @@ public int run() {
+                    try (Scanner scanner = parseInModule.getScanner()) {
+                        FileUtil files = this.files.get();
+                        File scannerFile = files.resolveTemp("scanner.l");
+                        File parserFile = files.resolveTemp("parser.y");
+                        scanner.getStandaloneScanner(scannerFile);
+                        KSyntax2Bison.getParser(parseInModule.getParsingModule(), scanner, sort, parserFile);
+                        int exit = files.getProcessBuilder()
+                          .directory(files.resolveTemp("."))
+                          .command("flex", "-w", scannerFile.getAbsolutePath())
+                          .inheritIO()
+                          .start()
+                          .waitFor();
+                        if (exit != 0) {
+                            throw KEMException.internalError("flex returned nonzero exit code: " + exit + "\n");
+                        }
+                        exit = files.getProcessBuilder()
+                          .directory(files.resolveTemp("."))
+                          .command("bison", "-d", "-Wno-other", "-Wno-conflicts-sr", "-Wno-conflicts-rr", parserFile.getAbsolutePath())
+                          .inheritIO()
+                          .start()
+                          .waitFor();
+                        if (exit != 0) {
+                            throw KEMException.internalError("bison returned nonzero exit code: " + exit + "\n");
+                        }
+                        exit = files.getProcessBuilder()
+                          .command("gcc",
+                              files.resolveKBase("include/cparser/main.c").getAbsolutePath(),
+                              files.resolveTemp("lex.yy.c").getAbsolutePath(),
+                              files.resolveTemp("parser.tab.c").getAbsolutePath(),
+                              "-iquote", files.resolveTemp(".").getAbsolutePath(),
+                              "-iquote", files.resolveKBase("include/cparser").getAbsolutePath(),
+                              "-o", outputFile.getAbsolutePath())
+                          .inheritIO()
+                          .start()
+                          .waitFor();
+                        if (exit != 0) {
+                            throw KEMException.internalError("gcc returned nonzero exit code: " + exit + "\n");
+                        }
+                    } catch(IOException | InterruptedException e) {
+                      throw KEMException.internalError("Failed to execute process.", e);
+                    }
+                }
+            } else {
+              K parsed = kread.prettyRead(parsingMod, sort, def, source, FileUtil.read(stringToParse));
+
+              if (options.expandMacros) {
+                  parsed = ExpandMacros.forNonSentences(unparsingMod, files.get(), def.kompileOptions, false).expand(parsed);
+              }
```

Comment:
Wow, the logic here got a lot more complicated.

I think having `kast --gen-parser` doesn't make much sense. `kompile --backend bison` or `kompile --emit-bison-parser` makes more sense to me. I also agree that the generated parser should go in the `kompiled` directory, and should form a standalone binary/executable which the user can call directly (without having to invoke immediately via `kast`).

If you absolutely must keep it here, put all the bison generation stuff it its own method and call it under the `if (options.genParser)` instead of inlining it, so that it's easier to see the overall logic of this main method.

It would also make sense to me to add `kast --use-bison-parser` and `krun --use-bison-parser`, which would use the parser emitted during `kompile`.

Reply:
Or, since `kompile --emit-bison-parser` will make a standalone executable, should be able to say `krun --parser path/to/definition-kompiled/bison-parser`, and no more options are needed?

Reply:
You cannot say `krun --parser` because krun currently doesn't know how to deserialize and execute something that has already been translated into kore. But since the whole point of this is to /not/ invoke krun at all, I think that's fine.